src/plugins/auto_batch/tests/functional/CMakeLists.txt

```diff
@@ -0,0 +1,27 @@
+# Copyright (C) 2018-2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set(TARGET_NAME ov_auto_batch_func_tests)
+
+set(SHARED_HEADERS_DIR "${OpenVINO_SOURCE_DIR}/src/tests/functional/plugin/shared/include")
+addIeTargetTest(
+        NAME
+            ${TARGET_NAME}
+        ROOT
+            ${CMAKE_CURRENT_SOURCE_DIR}
+        INCLUDES
+            ${CMAKE_CURRENT_SOURCE_DIR}
+            ${SHARED_HEADERS_DIR}
+        LINK_LIBRARIES
+            funcSharedTests
+        DEPENDENCIES
+            openvino_auto_batch_plugin
+        ADD_CPPLINT
+        LABELS
+            Auto_Batch
+)
+if (ENABLE_INTEL_GPU)
+    target_compile_definitions(${TARGET_NAME} PRIVATE HAVE_GPU_PLUGIN)
```

Comment:
@ilya-lavrenov do you know is there any unified method to decide which hw plugin should be tested or not? For example, the openvino package is built with gpu plugin enabled, but CI docker doesn't want to run any gpu plugin test case, is there any global control mechanism?

Reply:
I don't know about such unified methods. You need in runtime understand whether specific device is here or not.. 
or maybe configure test binary with -d. So, you can run `ov_auto_func_tests -D GPU`

Reply:
I prefer runtime checking, how about get_available_device() ?

Reply:
Update to runtime decide whether run specific plugin tests or not.

Reply:
BTW, it's even better to avoid usage of any HW plugin and test everything on MOCK objects, like it's done in caching tests

What do you think?

Reply:
I think we can have 2 auto_batch test samples like cpu/gpu/gna plugin:
1. functional test: ov_auto_batch_func_test, which will use actual plugin to run test cases; 
2. unit test: ov_auto_batch_unit_test, which will use MOCK object to test auto_batch plugin.

Considered we still don't have unit test for auto_batch plugin, so firstly we create this functional test sample, and then create another separated PR for unit test with MOCK objects.

Reply:
![image](https://user-images.githubusercontent.com/31196718/212356274-c1166c88-b242-444e-b521-b69b63ed782a.png)

Reply:
I agree that we need two types of tests, but:
- We need AUTO_BATCH (but it's applicable for other umbrella plugins) tests, which checks auto-batch only on all levels - on unit test level to check plugin internals, also it should cover AUTO_BATCH integration and invocation as a part of Core (we have some logic which checks whether auto-batch is applicable).
- We also need integration tests (behavior tests in out current notation), which are instantiated in plugin functional tests to check whether plugin is compatible with AUTO_BACTH (or any other umbrella plugin like HETERO). These tests don't test AUTO_BACTH, they test plugin. Having these tests is also important for Conformance testing to showcase that ANY plugin compliant or not with specific OpenVINO feature (auto batching, hetero execution, etc). For AUTO_BATCH it implies having optimal_batch_size property check for plugin, maybe support of model with batch > 1. But it should assume that such tests should test all operations with all possible batch sizes, because we should it as a prerequisite for AUTO_BATCH, it's not an integration point. 

Currently (in current master), we have only second type of test for AUTO_BATCH (for multi / auto we have unit tests and it's great). And as a part of this task we need to split these cases between 2 buckets: 
- does a test check whether ANY plugin can be run under AUTO_BATCH? 
- does a test is needed to test AUTO_BATCH itself?
Likely, we are OK to have some duplication here

Reply:
Type second test above should contain integration tests:
    1). Test other plugin whether is compatible with AUTO_BATCH: such as "GPU", "HETRO:BATCH"
    2). Test whether plugin can run under AUTO_BATCH,  such as "BATCH:GPU(16)", "BATCH:CPU(4)"

For Type first test, there will be another WIP PR(https://github.com/openvinotoolkit/openvino/pull/15211) to cover:
    1) AUTO_BATCH itself internal tests
    2) Core level test for AUTO_BATCH

Reply:
> Type second test above should contain integration tests:

Integration tests should be a part of plugin tests, IMO.
As I wrote, such tests should also be integrated into conformance to understand whether plugin is compatible with AUTO_BATCH?

Reply:
I'm not sure the accurate scope of integration test and conformance test, could you give some clarification?

Reply:
@ilya-lavrenov Could we handle AUTO_BATCH internal test and conformance test in separated PR?